File: languages/java/app-encryption/src/main/java/com/godaddy/asherah/crypto/bufferutils/ManagedBufferUtils.java
package com.godaddy.asherah.crypto.bufferutils;

import org.bouncycastle.util.Arrays;

public final class ManagedBufferUtils {
  private ManagedBufferUtils() { }

  public static void wipeByteArray(final byte[] sensitiveData) {
    Arrays.clear(sensitiveData);
  }

}

File: languages/csharp/AppEncryption/Crypto/BufferUtils/ManagedBufferUtils.cs
using System.Runtime.CompilerServices;
using Org.BouncyCastle.Utilities;

namespace GoDaddy.Asherah.Crypto.BufferUtils
{
    public static class ManagedBufferUtils
    {
        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
        public static void WipeByteArray(byte[] sensitiveData)
        {
            // NoOptimize to prevent the optimizer from deciding this call is unnecessary
            // NoInlining to prevent the inliner from forgetting that the method was no-optimize
            Arrays.Fill(sensitiveData, 0);
        }
    }
}
